src/axom/mir/EquiZAlgorithm.hpp

```diff
@@ -0,0 +1,1398 @@
+   * \param matsetView The matset view to use for the input data.
+   */
+  EquiZAlgorithm(const TopologyView &topoView,
+                 const CoordsetView &coordsetView,
+                 const MatsetView &matsetView)
+    : axom::mir::MIRAlgorithm()
+    , m_topologyView(topoView)
+    , m_coordsetView(coordsetView)
+    , m_matsetView(matsetView)
+  { }
+
+  /// Destructor
+  virtual ~EquiZAlgorithm() = default;
+
+// The following members are protected (unless using CUDA)
+#if !defined(__CUDACC__)
+protected:
+#endif
+
+#if defined(AXOM_EQUIZ_DEBUG)
+  void printNode(const conduit::Node &n) const
+  {
+    conduit::Node options;
+    options["num_children_threshold"] = 10000;
+    options["num_elements_threshold"] = 10000;
+    n.to_summary_string_stream(std::cout, options);
+  }
+#endif
+
+  /*!
+   * \brief Perform material interface reconstruction on a single domain.
+   *
+   * \param[in] n_topo The Conduit node containing the topology that will be used for MIR.
+   * \param[in] n_coordset The Conduit node containing the coordset.
+   * \param[in] n_fields The Conduit node containing the fields.
+   * \param[in] n_matset The Conduit node containing the matset.
+   * \param[in] n_options The Conduit node containing the options that help govern MIR execution.
+   *
+   * \param[out] n_newTopo A node that will contain the new clipped topology.
+   * \param[out] n_newCoordset A node that will contain the new coordset for the clipped topology.
+   * \param[out] n_newFields A node that will contain the new fields for the clipped topology.
+   * \param[out] n_newMatset A Conduit node that will contain the new matset.
+   * 
+   */
+  virtual void executeDomain(const conduit::Node &n_topo,
+                             const conduit::Node &n_coordset,
+                             const conduit::Node &n_fields,
+                             const conduit::Node &n_matset,
+                             const conduit::Node &n_options,
+                             conduit::Node &n_newTopo,
+                             conduit::Node &n_newCoordset,
+                             conduit::Node &n_newFields,
+                             conduit::Node &n_newMatset) override
+  {
+    namespace bputils = axom::mir::utilities::blueprint;
+    AXOM_ANNOTATE_SCOPE("EquizAlgorithm");
+
+    // Copy the options.
+    conduit::Node n_options_copy;
+    bputils::copy<ExecSpace>(n_options_copy, n_options);
+    n_options_copy["topology"] = n_topo.name();
+
+#if defined(AXOM_EQUIZ_DEBUG)
+    // Save the MIR input.
+    conduit::Node n_tmpInput;
+    n_tmpInput[n_topo.path()].set_external(n_topo);
+    n_tmpInput[n_coordset.path()].set_external(n_coordset);
+    n_tmpInput[n_fields.path()].set_external(n_fields);
+    n_tmpInput[n_matset.path()].set_external(n_matset);
+    conduit::relay::io::blueprint::save_mesh(n_tmpInput,
+                                             "debug_equiz_input",
+                                             "hdf5");
```

Comment:
Note: "hdf5" is only available when `AXOM_USE_HDF5` is defined

Reply:
I added guards for ``AXOM_USE_HDF5``